File: buzu/src/test/java/br/net/buzu/sample/ppl/StaticPerson.java
package br.net.buzu.sample.ppl;

import br.net.buzu.java.annotation.PplMetadata;
import br.net.buzu.java.model.Subtype;

/**
 *
 *
 * @author Douglas Siviotti
 * @since 1.0
 *
 */
@PplMetadata(maxOccurs=1)
public class StaticPerson {

	@PplMetadata(name="personName", size=10, maxOccurs=1, subtype=Subtype.CHAR)
	private String name;
	@PplMetadata(name="personAge", size=10, maxOccurs=1, subtype=Subtype.INTEGER)
	private Integer age;
	@PplMetadata(name="personCity", size=10, maxOccurs=1, subtype=Subtype.CHAR)
	private String city;

	public StaticPerson() {

	}

	public StaticPerson(String name, int age, String city) {
		super();
		this.name = name;
		this.age = age;
		this.city = city;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public Integer getAge() {
		return age;
	}

	public void setAge(Integer age) {
		this.age = age;
	}

	public String getCity() {
		return city;
	}

	public void setCity(String city) {
		this.city = city;
	}

	@Override
	public String toString() {
		return "Person [name=" + name + ", age=" + age + ", city=" + city + "]";
	}


}

File: ppl-lib/src/test/kotlin/br/net/buzu/sample/ppl/StaticPerson.kt
package br.net.buzu.sample.ppl

import br.net.buzu.annotation.PplMetadata
import br.net.buzu.model.Subtype
import br.net.buzu.sample.pojo.Person

/**
 *
 *
 * @author Douglas Siviotti
 * @since 1.0
 */
@PplMetadata(maxOccurs = 1)
class StaticPerson {

    @PplMetadata(name = "personName", size = 10, maxOccurs = 1, subtype = Subtype.CHAR)
    var name: String? = null
    @PplMetadata(name = "personAge", size = 10, maxOccurs = 1, subtype = Subtype.INTEGER)
    var age: Int? = null
    @PplMetadata(name = "personCity", size = 10, maxOccurs = 1, subtype = Subtype.CHAR)
    var city: String? = null

    constructor()

    constructor(name: String, age: Int, city: String) : super() {
        this.name = name
        this.age = age
        this.city = city
    }

    override fun toString(): String {
        return "Person [name=$name, age=$age, city=$city]"
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as StaticPerson

        if (name != other.name) return false
        if (age != other.age) return false
        if (city != other.city) return false

        return true
    }

    override fun hashCode(): Int {
        var result = name?.hashCode() ?: 0
        result = 31 * result + (age ?: 0)
        result = 31 * result + (city?.hashCode() ?: 0)
        return result
    }


    companion object {
        const val PERSON_NAME = "Ladybug   "
        const val PERSON_AGE = 15
        const val PERSON_CITY = "Paris     "
        const val PPL_STRING = "(personName:C10;personAge:I10;personCity:C10)Ladybug   0000000015Paris     "
        val INSTANCE = StaticPerson(PERSON_NAME, PERSON_AGE, PERSON_CITY)
    }
}
